### PR TITLE
go plugin: do not remove install /bin directory before build

### PR DESCRIPTION
**Pull request by cjp256**
Packages configured in a part's stage-packages are installed prior
to build and installed to $SNAPCRAFT_PART_INSTALL.  Cleaning
$SNAPCRAFT_PART_INSTALL/bin will purge any files that may have been
installed there.

Running go build should overwrite any previously installed files,
which will be OK for incremental builds.  However, as the go plugin
does not enable out_of_source_build, the entire build directory is
purged on every build anyways.

Add spread test for coverage.

Signed-off-by: Chris Patterson <chris.patterson@canonical.com>

Reported by @stgraber

### CODE REVIEW

**Comment by stgraber** [comment]
LGTM

File: snapcraft/plugins/go.py [hunk 1]
```diff
@@ -284,9 +284,7 @@ def _build_go_packages(self) -> None:
     def build(self) -> None:
         super().build()
 
-        # Clear the installation before continuing.
-        if os.path.exists(self._install_bin_dir):
-            shutil.rmtree(self._install_bin_dir)
+        # Ensure install directory exists.
         os.makedirs(self._install_bin_dir)
```
Comment (sergiusens) [inline snapcraft/plugins/go.py]: this no needs `exist_ok=True`

Comment (sergiusens) [reply snapcraft/plugins/go.py]: I fear this might break rebuilding

File: tests/spread/plugins/go/snaps/go-use-stage-packages/go-hello/main.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2014-2020 Canonical Ltd
```
Comment (sergiusens) [inline tests/spread/plugins/go/snaps/go-use-stage-packages/go-hello/main.go]:
```suggestion
 * Copyright (C) 2020 Canonical Ltd
```

**Pull request merged**